### ISSUE TITLE
Rewrite fstat tests

### ISSUE THREAD

**Issue by metan-ucw** [linux-test-project/ltp#517]
The testcases in syscalls/fstat are old, messy and duplicated and should be rewritten.

**Comment by mdoucha** [comment]
Christian Amann has rewritten the fstat tests back in June. Is this issue still relevant?

**Comment by pevik** [comment]
You're right, fstat tests have been fixed. fstatat and fstatfs should be also fixed, but that perhaps can be tracked with their own tickets.

**Comment by metan-ucw** [comment]
Yes these were rewritten, looking at the code the coverage could be increased a bit, the test for error returns covers just two out of many.

Also we do not check all the fields filled in by fstat, for instance there is a number of hardlinks that would require linking the file a few times in a setup then checking that the number is correct.

**Comment by iamkkushal** [comment]
### Yet to be covered:

The st_blocks should be more or less equal size/512

### Inputs from @metan-ucw on how to do the same:

For large enough files the size/512 rounded up should be close to the
number of blocks (there would be usually a few more blocks preallocated
for the file than the actuall size is). Filesystems tend to allocate
more than one block when file needs to be extended, it looks like common
value is PAGE_SIZE/512. So I guess that a reasonable test would create a
few files somewhere between half a megabyte and a few megabytes in size
and then check that the size/blocks ratio is somewhere between 512 and
512-epsilon. As long as the file size is order of magnitude larger than
PAGE_SIZE it should be safe to set the epsion to 10% of the blocksize
i.e. 51.
